llvm/test/Transforms/Coroutines/coro-split-musttail.ll

```diff
@@ -40,10 +38,8 @@ exit:
 
 ; Verify that in the resume part resume call is marked with musttail.
 ; CHECK-LABEL: @f.resume(
-; CHECK: %[[addr2:.+]] = call ptr @llvm.coro.subfn.addr(ptr null, i8 0)
-; NOPGO-NEXT: musttail call fastcc void %[[addr2]](ptr null)
-; PGO: call void @llvm.instrprof
```

Comment:
These `llvm.instrprof.value.profile` calls between the `llvm.coro.subfn.addr` call and tail call disappear with my patch. That's because the test runs `pgo-instr-gen` before `coro-split`. Is that representative of how the normal pipeline looks?

If the instrumentation is done after coro-split, it will see the indirect call and add the `llvm.instrprof.value.profile` as expected.

Reply:
I looked into this, and I don't think my patch changes the situation for value profiling. PGO instrumentation is performed early. Before my patch it would see a call to `@llvm.coro.resume`, and now it will see `@llvm.coro.await.suspend.handle`. In neither case does it see the lowered indirect call to the resumed function, and so no `@llvm.instrprof.value.profile` call gets inserted. Supporting that would be a nice future improvement.